src/Evolution/Systems/GeneralizedHarmonic/Characteristics.cpp

```diff
@@ -0,0 +1,191 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+// IWYU pragma: no_include <array>
+
+#include "Evolution/Systems/GeneralizedHarmonic/Characteristics.hpp"
+#include "Characteristics.hpp"
+#include "DataStructures/DataBox/Prefixes.hpp" // IWYU pragma: keep
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/EagerMath/DotProduct.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp" // IWYU pragma: keep
+#include "Evolution/Systems/GeneralizedHarmonic/Tags.hpp"
+#include "PointwiseFunctions/GeneralRelativity/IndexManipulation.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+// IWYU pragma: no_forward_declare Tensor
+// IWYU pragma: no_forward_declare Tags::CharSpeed
+// IWYU pragma: no_forward_declare Tags::Psi
+// IWYU pragma: no_forward_declare Tags::Pi
+// IWYU pragma: no_forward_declare Tags::Phi
+// IWYU pragma: no_forward_declare Tags::UPsi
+// IWYU pragma: no_forward_declare Tags::UZero
+// IWYU pragma: no_forward_declare Tags::UMinus
+// IWYU pragma: no_forward_declare Tags::UPlus
```

Comment:
So I understand better, why do you not forward-declare these tags here?

Reply:
IWYU complains that `Tags::U*` are being used here explicitly, while their code has not been included so. Their code is read in by the compiler when it parses `#include "Evolution/Systems/GeneralizedHarmonic/Tags.hpp"` while reading in `"Evolution/Systems/GeneralizedHarmonic/Characteristics.hpp"`. Since `Characteristics.hpp` will always be read in before `Tags.hpp` even if the latter is included explicltly below, there is no actual need to include the latter here other than to fix IWYU's complaint. Hence I suppressed IWYU's warnings here.

Reply:
I think in this case we should just include Tags.hpp. This should make IWYU happy and is in accordance with the Google style guide.

Reply:
I agree with @prayush: this looks like a IWYU bug.  (Tags.hpp is included by the header corresponding to this file.)